### DISCUSSION TITLE
Error installing cs_tools (python 3.13)

### DISCUSSION THREAD

**Discussion by Malfacini** [thoughtspot/cs_tools#180] [General] [answered]
Hi Team,

I am trying to install cs_tools but I'm getting an error saying it "Failed building wheel for msgspec". I'm installing it in windows using powershell. I've attached the error log file.

Are you able to help me figure this out?

Thank you,
[cs_tools-bootstrap-error-tmp2ogws52p.log](https://github.com/user-attachments/files/17465360/cs_tools-bootstrap-error-tmp2ogws52p.log)

Doug M.

**Comment by boonhapus** [comment, marked as answer]
Hey @Malfacini , it looks like some of our transitive dependencies don't yet play nicely with python 3.13 . `msgspec` is trying to compile from source because they have yet to release 3.13-specific whl file.

![image](https://github.com/user-attachments/assets/8805674b-76aa-4a9f-a8e7-f688c5f3c095)

Would it be possible to downgrade to python 3.12 in the meantime?

Installers **[can be found here](https://www.python.org/downloads/release/python-3126/)** , your architecture probably is happiest with this specific one: **[Windows installer (64-bit)](https://www.python.org/ftp/python/3.12.6/python-3.12.6-amd64.exe)**

**Reply by Malfacini** [reply]
Thank you for the quick reply - that worked! cs_tools is installed.

**Reply by IanMiner** [reply]
I ran into this same issue but when I unstalled python 3.13 and installed 3.12 my cstools  install keeps checking for python 3.13 even though I removed it from the PATH. Any idea how to fix this?
<img width="862" alt="cstootserror" src="https://github.com/user-attachments/assets/2bf426e9-0e7e-46fb-b0a7-88e8542f2969">
